libs/frontend/feature-site-store/src/lib/use-site-source.ts

```diff
@@ -60,14 +67,93 @@ export const useSiteSource = (): IUseSiteSource => {
     setRestoredSite(restored)
   }
 
+  // The periodic fetch on the preview page will reset page states, such as
+  // input values and styles added via behavior code. Therefore, we should
+  // only replace the current site when there are changes in content that
+  // will affect the rendered output.
   const checkOutdated = async () => {
+    if (!apiSiteId.value) {
+      // Scratch site
+      await checkScratchSiteOutdated()
+    } else {
+      // API site
+      await checkApiSiteOutdated()
+    }
+  }
+
+  // Only replace the current site with a new one when content changes.
+  const checkScratchSiteOutdated = async () => {
+    const currentScratchSite = localStorage.getItem('scratch-site-store') ?? ''
+    if (oldScratchSite !== currentScratchSite) {
+      if (currentScratchSite) {
+        const moduleState = JSON.parse(currentScratchSite)
+        store.scratchSite.setSite(moduleState.state)
+      }
+      oldScratchSite = currentScratchSite
+
+      const restored = await siteStore.value.restore(site.value.updated_at)
+      if (restored) {
+        setRestoredSite(restored)
+        console.log('Site updated:', site.value.updated_at)
+      }
+    }
+  }
```

Comment:
This is the code that replaces the site with the new one on preview page when the site is edited in another page.
I have to compare the raw state in `localStorage` because this is the only shared value I can get in two different browser tabs. Although we have the `LocalStoragePlugin` in `scratch-site-store.ts`, the state in the store will still not be updated without refreshing the page, given that they are two separate browser tabs.